### DISCUSSION TITLE
Create a Backup point before merging

### DISCUSSION THREAD

**Discussion by roryashfordbentley** [wpengine/github-action-wpe-site-deploy#64] [Ideas]
Hi team,

We are in the process of setting up a couple of hundred sites using the WPE GitHub action and its working incredibly well!! 

Would it be possible in a future release to automatically trigger a restore point on WP Engine before code deployed via this GitHub action is merged to the live site? This would be incredibly useful in an emergency.

**Comment by groenroos** [comment]
Just searched for this myself, and was disappointed to see it wasn't available. Would definitely want a flag to turn this on (even if it might be a good idea to keep it off by default for performance reasons).

Now, it seems to be possible to kick off a backup via [the WPEngine API](https://wpengineapi.com/reference), so adding that manually as a GHA job step before this Action shouldn't be too difficult... however, the tricky bit might be then polling the backup status endpoint at a reasonable cadence to ensure the backup has finished before running the deployment.
